### ISSUE TITLE
Enhancement: [prefer-optional-chain] Suggest `!foo || !foo.bar` as a valid match for the rule

### ISSUE THREAD

**Issue by omril1** [typescript-eslint/typescript-eslint#5245]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-optional-chain/

### Description

I propose that the `prefer-optional-chain` rule should also check for `!foo || !foo.bar` the boolean alternative form of `!(foo && foo.bar)` to be suggested as `!foo?.bar`

### Fail

```typescript
!foo || !foo.bar;

!foo || !foo[bar];

!foo || !foo.bar || !foo.bar.baz || !foo.bar.baz();
```

### Pass

```typescript
!foo?.bar;

!foo?.[bar];

!foo?.bar?.baz?.();
```

### Additional Info

Willing to open a PR

**Comment by bradzacher** [comment]
Just a heads up - #5266 introduced some crashes for users, so we reverted the PR.
Generally we would look to fix forward, but because the bugs were discovered by users in the release week, we thought it best to revert it to fix for the upcoming release, and we can look to rebuild the PR without the bugs without a time pressure.

**Comment by omril1** [comment]
@bradzacher Thanks, I'm taking a look
